base/compiler/abstractinterpretation.jl

```diff
@@ -558,11 +586,17 @@ function abstract_call_method(interp::AbstractInterpreter, method::Method, @nosp
         sparams = recomputed[2]::SimpleVector
     end
 
-    rt, edge = typeinf_edge(interp, method, sig, sparams, sv)
+    rt, edge, edge_effects = typeinf_edge(interp, method, sig, sparams, sv)
     if edge === nothing
         edgecycle = edgelimited = true
     end
-    return MethodCallResult(rt, edgecycle, edgelimited, edge)
+    if edgecycle
```

Comment:
`merge_call_chain` / `resolve_call_cycle!` might be a more appropriate places to handle this merge? I am not entirely sure of the relationships these days between these different functions and flags (edgecycle, pclimitations, `resolve_call_cycle!`, and the self-recursion detection above). They can, and are supposed to diverge, when we encounter various unused frames in the process (such as during constant-prop or because the result type is unused here). Yet that does not quite work optimally now.

Reply:
That should work I believe.